### DISCUSSION TITLE
using setValue alongside a Controller for the same field

### DISCUSSION THREAD

**Discussion by jhsu** [react-hook-form/react-hook-form#3952] [Q&A] [answered]
Is it safe to use a Controller and setValue from useForm to control the same field?

Here's a codesandbox that has multiple ways to show values and set values: https://codesandbox.io/s/updating-form-values-8fc1e?file=/src/App.js

notice when you change values in the input, the value in the `Controller` doesn't change. Am I using these concepts correctly?

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
i can see a problem where you have `register` the same input at useEffect and controller. `name` should be unique.
